timemachine/fe/energy_decomposition.py

```diff
@@ -47,8 +48,12 @@ def get_batch_u_fns(
     batch_u_fns: List[Batch_u_fn] = []
     for p, pot in zip(params, pots):
 
-        def batch_u_fn(xs, boxes, pot_impl, pot_params):
-            coords = np.array([x for x in xs])
+        def batch_u_fn(xs: NDArray | StoredArrays, boxes, pot_impl, pot_params):
+            # If the coordinates are already an in-memory numpy array, don't create a copy
+            if isinstance(xs, StoredArrays):
+                coords = np.array(xs)
```

Comment:
nit: could this just use `np.asarray` to avoid the conditional logic here?

Reply:
Done in https://github.com/proteneer/timemachine/pull/1323/commits/49d2b274ef9d06ee3abe45bdaa324905358af258